core/domain/wipeout_service.py

```diff
@@ -109,11 +107,109 @@ def _delete_user_models(user_id):
     """
     for model_class in models.Registry.get_storage_model_classes(
             [models.NAMES.user]):
-        if (model_class.get_deletion_policy() !=
-                base_models.DELETION_POLICY.KEEP):
+        if (model_class.get_deletion_policy() not in
+                [base_models.DELETION_POLICY.KEEP,
+                 base_models.DELETION_POLICY.NOT_APPLICABLE]):
             model_class.apply_deletion_policy(user_id)
 
 
+def _hard_delete_explorations_and_collections(pending_deletion_model):
+    """Hard delete the exploration and collection models that are private and
+    solely owned by the user.
+
+    Args:
+        pending_deletion_model: PendingDeletionRequestModel.
+    """
+    exp_services.delete_explorations(
+        pending_deletion_model.id,
+        pending_deletion_model.exploration_ids,
+        force_deletion=True)
+    collection_services.delete_collections(
+        pending_deletion_model.id,
+        pending_deletion_model.collection_ids,
+        force_deletion=True)
+
+
+def _generate_activity_to_user_ids_mapping(activity_ids):
+    """Generate mapping from activity IDs to user IDs.
+
+    Args:
+        activity_ids: list(str). List of activity IDs for which to generate
+            new user IDs.
+
+    Returns:
+        dict(str, str). Mapping between the activity IDs and pseudonymized
```

Comment:
It's unclear how the user IDs map to the activity. Can you clarify? E.g. are they people who have played the activity, edited it, manage it, etc.